cli/remoteforward.go

```diff
@@ -0,0 +1,104 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"regexp"
+	"strconv"
+
+	gossh "golang.org/x/crypto/ssh"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/agent/agentssh"
+)
+
+// cookieAddr is a special net.Addr accepted by sshRemoteForward() which includes a
+// cookie which is written to the connection before forwarding.
+type cookieAddr struct {
+	net.Addr
+	cookie []byte
+}
+
+var remoteForwardRegex = regexp.MustCompile(`^(\d+):(.+):(\d+)$`)
```

Comment:
nit: add a comment to clarify what this should match

Reply:
> nit: add a comment to clarify what this should match

👍 

> This could also be done with a simple strings.Split()

Right, but with regexp we can check if ports are numbers :)